# ER Mean & Median Wait Times: SQL to Pandas
Calculating average and median wait times from emergency room records. Validating SQL findings with Python for accuracy. 

In [1]:
import pandas as pd

## Load ER Visits Data

In [2]:
er_visits = pd.read_csv("../data/raw/er_visits.csv") 

## Preview of ER Visits Table

In [3]:
er_visits.head()

,visit_id,patient_id,arrival_date,arrival_day,arrival_hour,severity_level,wait_time_minutes,length_of_stay_hours,admitted
0,1,329,11/30/2025,Sunday,19,Low,137,4.7,No
1,2,152,1/7/2025,Tuesday,2,High,57,11.6,Yes
2,3,10,11/2/2025,Sunday,5,High,36,8.8,Yes
3,4,283,8/17/2025,Sunday,14,High,10,12.9,No
4,5,333,2/19/2026,Thursday,22,Medium,56,11.0,Yes


## SQL to Pandas Validations: Mean & Median ER Wait Times

In [4]:
round(er_visits["wait_time_minutes"].mean(),2)

np.float64(74.31)

In [5]:
er_visits["wait_time_minutes"].median()

np.float64(67.0)

## Breakdown of the Validation Results 
The average ER wait time is 74.31 mintues, and the median wait time is 67 minutes. Both results validate the same findings from SQL. 